### PR TITLE
Deploying multiple gpgkey in one repo

### PR DESCRIPTION
**Pull request by teluq-pbrideau**
#### Pull Request (PR) description
The yum repository format should support entering multiple keys in the `gpgkey` entry, like the following entry for gitlab:
```
[gitlab_gitlab-ce]
name=gitlab_gitlab-ce
baseurl=https://packages.gitlab.com/gitlab/gitlab-ce/el/8/$basearch
repo_gpgcheck=1
gpgcheck=1
enabled=1
gpgkey=https://packages.gitlab.com/gitlab/gitlab-ce/gpgkey
       https://packages.gitlab.com/gitlab/gitlab-ce/gpgkey/gitlab-gitlab-ce-3D645A26AB9FBD22.pub.gpg
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt
metadata_expire=300
```
Currently, this module only deploy the first gpgkey it encounter.  This change allow to deploy multiples gpg keys (separated by any white spaces), for example:
```
class { 'yum' :
  managed_repos => ['gitlab'],
  gpgkeys       => {
    '/etc/pki/rpm-gpg/RPM-GPG-KEY-gitlab'                  => { source => 'https://packages.gitlab.com/gitlab/gitlab-ce/gpgkey' },
    '/etc/pki/rpm-gpg/RPM-GPG-KEY-gitlab-3D645A26AB9FBD22' => { source => 'https://packages.gitlab.com/gitlab/gitlab-ce/gpgkey/gitlab-gitlab-ce-3D645A26AB9FBD22.pub.gpg' },
  },
  repos         => {
    'gitlab' => {
      [...]
      'gpgkey' => 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-gitlab file:///etc/pki/rpm-gpg/RPM-GPG-KEY-gitlab-3D645A26AB9FBD22',
    },
}
```

### CODE REVIEW

**Comment by teluq-pbrideau** [comment]
I’m not sure, but does this PR fixes #116? It seems old

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -706,6 +706,25 @@
         it { is_expected.not_to contain_package('yum-utils') }
         it { is_expected.to contain_package('dnf-utils') }
       end
+
+      context 'when custom repos is set' do
+        let(:params) { { managed_repos: ['example'], repos: { example: { baseurl: 'https://example.com', gpgkey: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example' } }, gpgkeys: { '/etc/pki/rpm-gpg/RPM-GPG-KEY-example' => { 'source' => 'http://example.com/gpg' } } } }
```
Comment (jhoblitt) [inline spec/classes/init_spec.rb:711]: The line length is excessive. Please reformat.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -706,6 +706,25 @@
...
+
+        it { is_expected.to compile.with_all_deps }
+        it { is_expected.to have_yumrepo_resource_count(1) }
+        it { is_expected.to contain_yumrepo('example') }
+        it { is_expected.to contain_yum__gpgkey('/etc/pki/rpm-gpg/RPM-GPG-KEY-example') }
+      end
+
+      context 'when custom repos with multiple gpgkeys is set' do
+        let(:params) { { managed_repos: ['example'], repos: { example: { baseurl: 'https://example.com', gpgkey: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example file:///etc/pki/rpm-gpg/RPM-GPG-KEY-example2', } }, gpgkeys: { '/etc/pki/rpm-gpg/RPM-GPG-KEY-example' => { 'source' => 'http://example.com/gpg' }, '/etc/pki/rpm-gpg/RPM-GPG-KEY-example2' => { 'source' => 'http://example.com/gpg2' } } } }
```
Comment (jhoblitt) [inline spec/classes/init_spec.rb:720]: The line length is excessive. Please reformat.

File: manifests/init.pp [hunk 3]
```diff
@@ -152,18 +152,20 @@
...
-          } # end if gpgkey is a file:// resource
+          $matches = $attributes['gpgkey'].split(/\s/).match('^file://(.*)$')
+          $matches.each |$match| {
+            if $match {
+              $gpgkey = $match[1]
+              if $gpgkey =~ Stdlib::AbsolutePath and $gpgkey in $gpgkeys {
+                if !defined(Yum::Gpgkey[$gpgkey]) {
+                  yum::gpgkey { $gpgkey:
+                    *      => $gpgkeys[$gpgkey],
+                    before => Package[$utils_package_name],  # GPG Keys for any managed repository need to be installed before we attempt to install any packages.
```
Comment (jhoblitt) [inline manifests/init.pp:163]: I realize this is copied from the existing code but the trailing comment makes this line excessively long.

**Review by jhoblitt — CHANGES_REQUESTED** [review]
This seems reasonable to me. I think it is ready for merge with some minor reformatting.

File: manifests/init.pp (hunk 3 above)
Comment (jhoblitt) [reply manifests/init.pp]: I probably would have put it on the line above but I think it is acceptable as-is.

**Review by jhoblitt — APPROVED** [review]

**Comment by jhoblitt** [comment]
This LGTM to me now. Thank you for contributing.

**Pull request merged**